### PR TITLE
vkd3d: Add VKD3D_SHADER_MODEL override.

### PR DESCRIPTION
**Pull request by CME42**
This makes it possible to run tlou1 on pascal, but may cause unexpected results.

### CODE REVIEW

File: libs/vkd3d/device.c [hunk 1]
```diff
@@ -6669,6 +6669,50 @@ static void d3d12_device_caps_init_feature_level(struct d3d12_device *device)
...
+    shader_models[] =
+    {
+        { "5.1", D3D_SHADER_MODEL_5_1 },
+        { "6.0", D3D_SHADER_MODEL_6_0 },
+        { "6.1", D3D_SHADER_MODEL_6_1 },
+        { "6.2", D3D_SHADER_MODEL_6_2 },
+        { "6.3", D3D_SHADER_MODEL_6_3 },
+        { "6.4", D3D_SHADER_MODEL_6_4 },
+        { "6.5", D3D_SHADER_MODEL_6_5 },
+        { "6.6", D3D_SHADER_MODEL_6_6 },
```
[the remark this answers is not in the document]
Comment (CME42) [reply libs/vkd3d/device.c:6691]: seems like D3D_SHADER_MODEL_6_7 is not in the enum yet, shall i just add it?

**Comment by CME42** [comment]
> Can we make this 6_1 to match the feature level formatting?

I thought VKD3D_SHADER_MODEL=6_6 looks ugly so I made dots out of it 🐸 But I don't really mind, can change that to underscore.

File: libs/vkd3d/device.c [hunk 2]
```diff
@@ -6669,6 +6669,50 @@ static void d3d12_device_caps_init_feature_level(struct d3d12_device *device)
     TRACE("Max feature level: %#x.\n", caps->max_feature_level);
 }
 
+static void d3d12_device_caps_shader_model_override(struct d3d12_device *device) {
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:6672]: Brace placement

File: libs/vkd3d/device.c [hunk 3]
```diff
@@ -6669,6 +6669,50 @@ static void d3d12_device_caps_init_feature_level(struct d3d12_device *device)
...
+    D3D_SHADER_MODEL sm_override = (D3D_SHADER_MODEL)0;
+    char sm_string[VKD3D_PATH_MAX];
+    unsigned int i;
+
+    static const struct
+    {
+        const char* string;
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:6679]: `const char *string`

File: libs/vkd3d/device.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (HansKristian-Work) [reply libs/vkd3d/device.c:6691]: Add it to IDL, yes.

File: libs/vkd3d/device.c [hunk 4]
```diff
@@ -6669,6 +6669,50 @@ static void d3d12_device_caps_init_feature_level(struct d3d12_device *device)
...
+        {
+            sm_override = shader_models[i].shader_model;
+            break;
+        }
+    }
+
+    if (!sm_override)
+    {
+        WARN("Unrecognized shader model %s.\n", sm_string);
+    } else {
```
Comment (HansKristian-Work) [inline libs/vkd3d/device.c:6709]:
```
if ()
{
}
else
{
}
```

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]

**Comment by HansKristian-Work** [comment]
> Can we make this 6_1 to match the feature level formatting?

It also matches how you invoke DXC. `-Tcs_6_4` for example.

**Review by HansKristian-Work — CHANGES_REQUESTED** [review]
Smol nit, then lgtm. Before taking this, do you consent to this being relicensed to MIT as well?

**Review by HansKristian-Work — APPROVED** [review]
LGTM

**Pull request merged**